### ISSUE TITLE
How to categorize images with multiple target elements using yolo object detection methods

### ISSUE THREAD

**Issue by LXXiaogege** [ultralytics/yolov5#12559]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I want to use yolo to do a website identification task: specifically, to identify the type of website based on the objects in the snapshot of the website, for example, if there are a lot of betting elements in the website, then I think it is a betting website, but there are usually more than one element in a picture, so I need to make some kind of rules to post-process the results of yolo to get the type of website. The method I came up with is:
1, design rules
2、Intervene a svm or lr classifier in the result recognized by yolo
3、Modify the network structure and loss function of yolo outputs
Is there any ready solution for this type of problem or any suggestion ? thanks

**Comment by glenn-jocher** [comment]
@LXXiaogege hi there!👋 To categorize images with multiple target elements using YOLO, you can post-process the results by designing specific rules or integrating a classifier like SVM or LR. Alternatively, you can customize the network structure and loss function of YOLO outputs. For more details, check the Ultralytics Docs at https://docs.ultralytics.com/yolov5/. Good luck!

**Comment by LXXiaogege** [comment]
> @LXXiaogege hi there!👋 To categorize images with multiple target elements using YOLO, you can post-process the results by designing specific rules or integrating a classifier like SVM or LR. Alternatively, you can customize the network structure and loss function of YOLO outputs. For more details, check the Ultralytics Docs at https://docs.ultralytics.com/yolov5/. Good luck!

You're just repeating my thoughts, bad guy， Another idea I have is similar to bert, use your pre-trained yolo model, or use my fine-tuned yolo model, freeze the model parameters, use  MLP classifier at the back, and then train, I don't know if it is feasible

**Comment by glenn-jocher** [comment]
@LXXiaogege Thanks for sharing your thoughts! Using a pre-trained YOLO model with frozen parameters and an MLP classifier is indeed feasible. It's a technique commonly used for transfer learning. Feel free to experiment with it and see how it performs for your specific task. Good luck!
